src/utils/internal.ts

```diff
@@ -1891,3 +1898,75 @@ export function areSameClaims(
 
   return ClaimType[type] === claim.type;
 }
+
+/**
+ * @hidden
+ */
+export function throwIfPendingAuthorizationExists(params: {
+  authorizationRequests: AuthorizationRequest[];
+  message: string;
+  authorization: Partial<Authorization>;
+  issuer?: Identity;
+  target?: string | Identity;
+}): void {
+  const {
+    authorizationRequests,
+    message,
+    authorization,
+    target: targetToCheck,
+    issuer: issuerToCheck,
+  } = params;
+
+  if (authorizationRequests.length === 0) {
+    return;
+  }
+
+  const hasPendingAuth = !!authorizationRequests.find(authorizationRequest => {
+    const { issuer, target, data } = authorizationRequest;
+
+    if (authorizationRequest.isExpired()) {
+      return false;
+    }
+
+    if (targetToCheck && signerToString(target) !== signerToString(targetToCheck)) {
+      return false;
+    }
+
+    if (issuerToCheck && signerToString(issuer) !== signerToString(issuerToCheck)) {
+      return false;
+    }
+
+    if (authorization.type && data.type !== authorization.type) {
+      return false;
+    }
+
+    if (authorization.type === AuthorizationType.PortfolioCustody && authorization.value) {
+      const authorizationData = data as { value: NumberedPortfolio | DefaultPortfolio };
+
+      return authorizationData.value.isEqual(authorization.value);
+    }
+
+    if (authorization.type === AuthorizationType.AttestPrimaryKeyRotation && authorization.value) {
+      const authorizationData = data as AttestPrimaryKeyRotationAuthorizationData;
+
+      return authorizationData.value.isEqual(authorization.value);
+    }
+
+    // last checks for authorizations that have string values
+    const { value } = authorization as GenericAuthorizationData;
+    const { value: authorizationValue } = data as GenericAuthorizationData;
+
+    if (value && value !== authorizationValue) {
+      return false;
+    }
+
+    return true;
+  });
+
+  if (hasPendingAuth) {
+    throw new PolymeshError({
+      code: ErrorCode.NoDataChange,
+      message,
+    });
```

Comment:
There is a `data` property you can add.

I think there can be some additional context you can add into it to help with debugging if it ever gets thrown.

```suggestion
      message,
      data: {target, issuer, authorizationType, authorizationId}
    });
```